### PR TITLE
Budget app

### PR DESCRIPTION
**Pull request by kit0-0**
# Description

This pull request encompasses a series of fundamental changes aimed at establishing the initial structure and functionality of the Ruby on Rails application. Here is a breakdown of the modifications:

### Setting Up
- Created a new Rails application configured with a PostgreSQL database for development.
- Integrated a linter and Rubocop for code styling consistency.
- Established the necessary models and their associations.
- Configured routes to define the application's navigation.

### Authentication
- Integrated Devise to handle authentication within the application.

### Pages and Functionality
- Introduced a splash page providing a welcoming interface for login and sign-up.
- Implemented dedicated login and sign-up pages.
- Developed the categories page.
- Implemented the expenses page.

### Testing
- Implemented comprehensive integration tests and model tests to ensure the correctness of the application.

### CODE REVIEW

File: app/views/categories/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            <div class="text-lg font-weight-bold"><%= category.name %></div>
+            <div class="text-lg font-weight-bold">$<%= format('%.2f', category.expenses.sum(:amount)) %></div>
+          <% end %>
+        <% end %>
+      </ul>
+    <% end %>
+
+    <div class="add-category-container d-flex justify-content-center px-3 px-md-5 py-3">
+      <%= link_to "ADD CATEGORY", new_category_path, class: 'btn btn-outline-light btn-lg' %>
+    </div>
```
Comment (TadesseDev) [inline app/views/categories/index.html.erb]: - [ ] Dear @kit0-0, I'm already impressed with your app, Although the layout is great I notice we can add something minor to follow up along side the provided template. I noticed your `add a new category` and `Add expense ` buttons are not fixed to the bottom of the screen. To follow the design guideline please make sure this button is stick to the bottom of the page.

### Your design 
Screenshot 2023-10-25 at 10 00 55 at night

### Original design 
Screenshot 2023-10-25 at 10 10 17 at night

File: app/controllers/categories_controller.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    @category = Category.new
+    @options = ICON_OPTIONS
+  end
+
+  def create
+    @category = current_user.categories.build(categories_params)
+
+    if @category.save
+      redirect_to categories_path, notice: 'Expense category was successfully created.'
+    else
```
Comment (TadesseDev) [inline app/controllers/categories_controller.rb]: - [x] dear @kit0-0, please keep in mind we need to keep our inputs validated and make sure our app does not crash when an invalid input is submitted. This validation is done both from the HTML and the Rails side. currently, when I create a category by setting the name without selecting an icon, an error is thrown, and the app crashes. Please take a look into this. 

### The issue 

https://github.com/kit0-0/budget/assets/69077061/ec3bf1a1-4d29-4e5e-90a2-6c5e536ede27

- [x] This issue persists when I try to create an expense without checking any of the checkboxes, Please consider addressing this too.

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @kit0-0 
### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x] The app is deployed on the web. ✅ 
 - [x] Good job overall following the template design ✅
 - [x] You have written a clean code ✅
 - [x]  Most of the requirement has been meet. ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- N/A
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KanzaTahreem — APPROVED** [review]
# Approved 🙌

Hi @kit0-0 

Good job implementing the suggested changes 💪 **_Your project is COMPLETE! There is nothing else to say other than... it's time to merge it! 🚀 Congratulations!_**



## Highlights 🎯
There are some excellent points that you have made perfectly. Keep up the great work! 👏

-  ✅ Styling is looking great
-  ✅ Implemented suggested changes 
-  ✅ Your README looks good
-  ✅ Linters are working properly

## Optional suggestions ❕
N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**